File: Easy/PrimePalindrome.java
import java.io.*;
public class Main {
    public static String solve(String input) {
        for(Integer i = Integer.parseInt(input) - 1; i >= 0; i--) {
            if(isPrime(i) && isPalindrome(i)) {
                return i.toString();
            }
        }

        return "1";
    }

    private static boolean isPalindrome(Integer number) {
        String numberString = number.toString();
        for(Integer i = 0; i < numberString.length() / 2; i++) {
            if(numberString.charAt(i) != numberString.charAt((numberString.length() - 1) - i)) {
                return false;
            }
        }

        return true;
    }

    private static boolean isPrime(Integer number) {
        for(Integer i = 2; i < number / 2; i++) {
            if(number % i == 0) {
                return false;
            }
        }

        return true;
    }
    
    public static void main (String[] args) throws IOException {
        //File file = new File(args[0]);
        //BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        //while ((line = buffer.readLine()) != null) {
            line = "1000";
            System.out.println(solve(line));
        //}
    }
}

File: Easy/PrimePalindrome.cpp
#include <iostream>
#include <fstream>
#include <sstream>

using namespace std;

bool isPalindrome(int number) {
    int n = number;
    int reverse_number = 0;
    while (n > 0) {
        reverse_number = reverse_number * 10 + (n % 10);
        n /= 10;
    }

    return reverse_number == number;
}

bool isPrime(int number) {
    for(int i = 2; i < number / 2; i++) {
        if(number % i == 0) {
            return false;
        }
    }

    return true;
}

string solve(const string &input) {
    for(int i = stoi(input) - 1; i >= 0; i--) {
        if(isPrime(i) && isPalindrome(i)) {
            return to_string(i);
        }
    }

    return "1";
}

int main(int argc, const char * argv[]) {
    cout << solve("1000") << endl;
    return 0;
}
